### PR TITLE
fix otelcol_exporter_send_failed_requests count error

### LINKED ISSUE #7456: Metrics send_failed_requests is inaccurate/abnormal
**Describe the bug**

I found that regardless of whether in a local or other environment, when attempting to send data, Otel exporter always reports a number greater than 0 for send failed requests. This is puzzling. The number of send failed requests always matches the group of spans sent. For example, based on the information below, 3 group of spans were sent and 3 metric of send failed requests were generated. Would anyone be able to assist in identifying the root cause of this issue? Thank you

**Steps to reproduce**
1. start the collector locally
./otelcol-contrib --config config.yaml
```yaml
receivers:
  otlp/recv-1:
    protocols:
      grpc:
        endpoint: localhost:4317
exporters:
  logging:
processors:
service:
  pipelines:
    traces/agent:
      receivers: [otlp/recv-1]
      processors: []
      exporters: [logging]
```
2. use this [example](https://github.com/open-telemetry/opentelemetry-go/blob/main/example/otel-collector/main.go), modify the endpoint of #L63 to localhost:4317
3. go run main.go

**What did you expect to see?**
otelcol_exporter_send_failed_requests should be 0

**What did you see instead?**
I saw that the exporter sent a total of 11 divided into 3 groups, and there were 3 send failed reqeusts
image
image

**What version did you use?**
Version: v0.73.0, version > 0.63.0 has this problem

**What config did you use?**
Config: (e.g. the yaml config file)

**Environment**
OS: MacOS M1
Compiler(if manually compiled): go1.18 darwin/arm64

**Additional context**

### PR DESCRIPTION
**Pull request by HeavenTonight**
Signed-off-by: guiyong.ou <guiyong.ou@daocloud.io>
**Description:** <Describe what has changed. 
fix otelcol_exporter_send_failed_requests count error

**Link to tracking Issue:** <Issue number if applicable>
fixed #7456

### CODE REVIEW

**Comment by jpkrohling** [comment]
@TylerHelmuth, you've been involved in the discussion at the original issue, would you mind reviewing this one?

File: .chloggen/fix-send_failed_requests-counter.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: obsreport
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: fix send_failed_requests counter error,because brought up again in the PR of #6346
```
Comment (TylerHelmuth) [inline .chloggen/fix-send_failed_requests-counter.yaml:8]:
```suggestion
note: fix issue where send_failed_requests counter was reporting an incorrect value.
```

**Review by TylerHelmuth — APPROVED** [review]
@HeavenTonight thanks for taking care of this.

**Comment by TylerHelmuth** [comment]
@jpkrohling can you approve the workflows?  I don't have permissions in this repo.

File: .chloggen/fix-send_failed_requests-counter.yaml (hunk 1 above)
Comment (HeavenTonight) [reply .chloggen/fix-send_failed_requests-counter.yaml:8]: > ```suggestion
> note: fix issue where send_failed_requests counter was reporting an incorrect value.
> ```

Thanks,i will change this!!

File: .chloggen/fix-send_failed_requests-counter.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: obsreport
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: fix issue where send_failed_requests counter was reporting an incorrect value.
+
+# One or more tracking issues or pull requests related to the change
+issues: [7456]
```
Comment (bogdandrutu) [inline .chloggen/fix-send_failed_requests-counter.yaml:11]: New line.

**Review by bogdandrutu — APPROVED** [review]

File: .chloggen/fix-send_failed_requests-counter.yaml (hunk 2 above)
Comment (HeavenTonight) [reply .chloggen/fix-send_failed_requests-counter.yaml:11]: > New line.

thanks for check！！

**Review by codeboten — APPROVED** [review]

**Pull request merged**